src/main/java/com/acciente/oacc/sql/internal/SQLAccessControlContext.java

```diff
@@ -927,7 +927,7 @@ private Resource __createResource(SQLConnection connection,
 
       // check to ensure that the specified external id does not already exist
       if (externalId != null && resourcePersister.resolveResourceByExternalId(connection, externalId) != null) {
-         throw new IllegalArgumentException("External id is not unique: " + externalId);
+         throw new IllegalArgumentException("External id is unique: " + externalId);
```

Comment:
Here we're checking if the method parameter `externalId` is already associated with an existing resource, before creating a new resource with that `externalId`.  
If this check fails (when `resolveResourceByExternalId()` returns a non-null value), the requested `externalId` for the new resource was **not** unique, because it's already associated with a resource.

In other words, this error message is trying to communicate that the `externalId` requested for the new resource is already in use for a previously created resource, hence it is not a unique external identifier.